File: todo_list_back/src/main/java/com/example/todo_list_back/entity/Todo.java
package com.example.todo_list_back.entity;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;

@Entity
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Todo {
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Long id;

    private String title;

    @Column(columnDefinition = "text")
    private String description;

    private boolean completed;

    @Temporal(TemporalType.TIMESTAMP)
    @Column(name = "created_at")
    private LocalDateTime createdAt;
    @Temporal(TemporalType.TIMESTAMP)
    @Column(name = "updated_at")
    private LocalDateTime updatedAt;
}

File: todo-list-front/src/formik/initial-value/todo.js


export const TodoInitialValues = (userId) => {
    return{
        title:"",
        description: "",
        completed: false,
        userId
    }
};

export const TodoUpdateInitialValues = (todo) => {
    return {
        title: todo.title,
        description: todo.description,
        completed: todo.completed,
        userId: todo.userId
    }
}